test/20_royalties_tests.ts

```diff
@@ -540,4 +550,216 @@ describe('HolographRoyalties Contract', async function () {
 
     it('should allow inherited contract to call fn');
   });
+
+  describe('Royalties Distribution Validation', () => {
+    const errorMargin = ethers.utils.parseUnits('2000', 'wei');
+
+    describe('A collection with 1 recipient', async () => {
+      const totalRoyalties = ethers.utils.parseEther('1');
+
+      before(async () => {
+        const data = (await royalties.populateTransaction.configurePayouts([owner.address], [10000])).data || '';
+        const tx = await l1.factory.connect(owner).adminCall(royalties.address, data);
+        await tx.wait();
+      });
+
+      it('should be able to withdraw all native token balance', async () => {
+        await owner.sendTransaction({
+          to: royalties.address,
+          value: totalRoyalties,
+        });
+
+        const accountBalanceBefore = await ethers.provider.getBalance(owner.address);
+        const contractBalanceBefore = await ethers.provider.getBalance(royalties.address);
+
+        const data = (await royalties.populateTransaction.getEthPayout()).data || '';
+        const tx = await l1.factory.connect(owner).adminCall(royalties.address, data);
+        await tx.wait();
+
+        const accountBalanceAfter = await ethers.provider.getBalance(owner.address);
+        const contractBalanceAfter = await ethers.provider.getBalance(royalties.address);
+
+        expect(contractBalanceAfter).to.be.lt(contractBalanceBefore);
+        expect(accountBalanceAfter).to.be.gt(accountBalanceBefore);
+      });
+
+      it('should be able to withdraw balance of an ERC20 token', async () => {
+        const ERC20 = await l1.holographErc20.attach(l1.sampleErc20Holographer.address);
+        const SAMPLEERC20 = await l1.sampleErc20.attach(l1.sampleErc20Holographer.address);
+
+        await SAMPLEERC20.mint(owner.address, totalRoyalties);
+        await ERC20.transfer(royalties.address, totalRoyalties);
+
+        const contractBalanceBefore = await ERC20.balanceOf(royalties.address);
+        const accountBalanceBefore = await ERC20.balanceOf(owner.address);
+
+        const data = (await royalties.populateTransaction.getTokenPayout(ERC20.address)).data || '';
+        const tx = await l1.factory.connect(owner).adminCall(royalties.address, data);
+        await tx.wait();
+
+        const accountBalanceAfter = await ERC20.balanceOf(owner.address);
+        const contractBalanceAfter = await ERC20.balanceOf(royalties.address);
+
+        expect(contractBalanceAfter).to.be.lt(contractBalanceBefore);
+        expect(accountBalanceAfter).to.be.gt(accountBalanceBefore);
```

Comment:
Again, we should be able to check specific values.

Reply:
yep updated to use equals instead